### PR TITLE
Setup inlining, mark all implicit value classes as final with private val

### PR DESCRIPTION
**Pull request by REDNBLACK**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -34,7 +34,7 @@ lazy val defaultSettings = List(
   libraryDependencies += compilerPlugin("com.olegpy"    %% "better-monadic-for" % "0.3.1"),
   libraryDependencies += scalatest,
   libraryDependencies ++=
-    List(
+    Seq(
```
Comment (Odomontois) [inline build.sbt]: What is accomplised here? Actually I can't see significant difference

File: concurrent/src/main/scala/tofu/concurrent/Daemon.scala [hunk 2]
```diff
@@ -88,11 +88,11 @@ object Daemon extends DaemonInstances {
     }
   }
 
-  implicit class DaemonOps[F[_], E, A](val self: Daemon[F, E, A]) extends AnyVal {
+  implicit final class DaemonOps[F[_], E, A](private val self: Daemon[F, E, A]) extends AnyVal {
```
Comment (Odomontois) [inline concurrent/src/main/scala/tofu/concurrent/Daemon.scala]: What it actually does. I've thought whole `AnyVal` buisness here is not allocate any Ops, so what difference `final` modifier does?

File: env/src/main/scala/tofu/env/EnvSpecializedFunctions.scala [hunk 3]
```diff
@@ -179,7 +179,7 @@ trait EnvSpecializedFunctions[E] {
   }
 
   object converters {
-    implicit class TaskAsEnv[A](val task: Task[A]) {
+    implicit class TaskAsEnv[A](private val task: Task[A]) {
```
Comment (Odomontois) [inline env/src/main/scala/tofu/env/EnvSpecializedFunctions.scala]: That's good catch

File: build.sbt (hunk 1 above)
Comment (REDNBLACK) [reply build.sbt]: I remember that author of sbt recommended usage of Seq, but i did it just for consistency ¯\_(ツ)_/¯

File: concurrent/src/main/scala/tofu/concurrent/Daemon.scala (hunk 2 above)
Comment (REDNBLACK) [reply concurrent/src/main/scala/tofu/concurrent/Daemon.scala]: final is for the better chance of inlining

**Comment by REDNBLACK** [comment]
@Odomontis One thing, do we care about bincompat, or planning to setup sbt-mima plugin for future releases of tofu? If yes, then we shouldn’t enable “opt:l:inline” (cross-method inlining).

File: concurrent/src/main/scala/tofu/concurrent/Daemon.scala (hunk 2 above)
Comment (Odomontois) [reply concurrent/src/main/scala/tofu/concurrent/Daemon.scala]: But those operations are never exposed as methods of classes, they should be inlined by the compiler itself

**Pull request merged**